### PR TITLE
add system variable tidb_analyze_skip_column_types

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What is changed, added or deleted? (Required)

Add system variable `tidb_analyze_skip_column_types`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/44653

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -839,6 +839,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 这个变量用于控制 TiDB 收集统计信息的行为。
 - 在 v5.3.0 及之后的版本中，该变量的默认值为 `2`，具体可参照[统计信息简介](/statistics.md)文档。如果从 v5.3.0 之前版本的集群升级至 v5.3.0 及之后的版本，`tidb_analyze_version` 的默认值不发生变化。
 
+### `tidb_analyze_skip_column_types` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值："json"
```
Comment (hfxsd) [inline system-variables.md:846]: 需要加上可选值

File: system-variables.md [hunk 2]
```diff
@@ -839,6 +839,13 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 这个变量表示在执行 ANALYZE 命令收集统计信息时，跳过哪些类型的列的统计信息收集。
```
Comment (hfxsd) [inline system-variables.md:847]:
```suggestion
- 该变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。
```

File: statistics.md [hunk 3]
```diff
@@ -244,6 +244,8 @@ SHOW COLUMN_STATS_USAGE WHERE db_name = 'test' AND table_name = 't' AND last_ana
 3 rows in set (0.00 sec)
 ```
 
+从 TiDB v7.2 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```
Comment (hfxsd) [inline statistics.md:247]:
```suggestion
从 TiDB v7.2.0 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```

**Review by hfxsd — APPROVED** [review]

File: statistics.md [hunk 4]
```diff
@@ -244,6 +244,8 @@ SHOW COLUMN_STATS_USAGE WHERE db_name = 'test' AND table_name = 't' AND last_ana
 3 rows in set (0.00 sec)
 ```
 
+从 TiDB v7.2.0 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```
Comment (ran-huang) [inline statistics.md:247]: 这一段放到这一小节的开头部分更合适，放在这里读者很容易漏掉，比如移到 L134 的注意里。
image

File: system-variables.md [hunk 5]
```diff
@@ -839,6 +839,60 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 这个变量用于控制 TiDB 收集统计信息的行为。
 - 在 v5.3.0 及之后的版本中，该变量的默认值为 `2`，具体可参照[统计信息简介](/statistics.md)文档。如果从 v5.3.0 之前版本的集群升级至 v5.3.0 及之后的版本，`tidb_analyze_version` 的默认值不发生变化。
 
+### `tidb_analyze_skip_column_types` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值："json,blob,mediumblob,longblob"
+- 可选值："json,blob,mediumblob,longblob,text,mediumtext,longtext"
+- 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。使用 `analyze table t columns c1, ..., cn` 语法时，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，则不会收集该列的统计信息。
```
Comment (ran-huang) [inline system-variables.md:848]:
```suggestion
- 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。使用 `ANALYZE TABLE t COLUMNS c1, ..., cn` 语法时，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，则不会收集该列的统计信息。
```

File: system-variables.md [hunk 6]
```diff
@@ -839,6 +839,60 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+```
+mysql> show create table t;
```
Comment (ran-huang) [inline system-variables.md:851]:
````suggestion
```sql
mysql> SHOW CREATE TABLE t;
````

File: system-variables.md [hunk 7]
```diff
@@ -839,6 +839,60 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+  `a` int(11) DEFAULT NULL,
+  `b` varchar(10) DEFAULT NULL,
+  `c` json DEFAULT NULL,
+  `d` blob DEFAULT NULL,
+  `e` longblob DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+
+mysql> select @@tidb_analyze_skip_column_types;
```
Comment (ran-huang) [inline system-variables.md:865]:
```suggestion
mysql> SELECT @@tidb_analyze_skip_column_types;
```

File: system-variables.md [hunk 8]
```diff
@@ -839,6 +839,60 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+| job_info                                                            |
++---------------------------------------------------------------------+
+| analyze table columns a, b with 256 buckets, 500 topn, 1 samplerate |
++---------------------------------------------------------------------+
+1 row in set (0.00 sec)
+
+mysql> analyze table t columns a, c;
+Query OK, 0 rows affected, 1 warning (0.04 sec)
+
+mysql> select job_info from mysql.analyze_jobs order by end_time desc limit 1;
```
Comment (ran-huang) [inline system-variables.md:887]:
```suggestion
mysql> ANALYZE TABLE t;
Query OK, 0 rows affected, 1 warning (0.05 sec)

mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
+---------------------------------------------------------------------+
| job_info                                                            |
+---------------------------------------------------------------------+
| analyze table columns a, b with 256 buckets, 500 topn, 1 samplerate |
+---------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> ANALYZE TABLE t COLUMNS a, c;
Query OK, 0 rows affected, 1 warning (0.04 sec)

mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
```

File: statistics.md (hunk 4 above)
Comment (hfxsd) [reply statistics.md]: done

File: statistics.md [hunk 9]
```diff
@@ -132,6 +132,7 @@ ANALYZE TABLE TableNameList [WITH NUM BUCKETS|TOPN|CMSKETCH DEPTH|CMSKETCH WIDTH
 > **注意：**
 >
 > 收集部分列的统计信息的功能仅适用于 `tidb_analyze_version = 2` 的情况。
+> 从 TiDB v7.2.0 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```
Comment (hfxsd) [inline statistics.md:135]:
```suggestion
> 从 TiDB v7.2.0 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。
```

Comment (hfxsd) [inline statistics.md:135]:
```suggestion
> 收集部分列的统计信息的功能仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入)  的情况。
> > 从 TiDB v7.2.0 起，TiDB 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```

File: system-variables.md [hunk 10]
```diff
@@ -839,6 +839,60 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 这个变量用于控制 TiDB 收集统计信息的行为。
 - 在 v5.3.0 及之后的版本中，该变量的默认值为 `2`，具体可参照[统计信息简介](/statistics.md)文档。如果从 v5.3.0 之前版本的集群升级至 v5.3.0 及之后的版本，`tidb_analyze_version` 的默认值不发生变化。
 
+### `tidb_analyze_skip_column_types` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值："json,blob,mediumblob,longblob"
+- 可选值："json,blob,mediumblob,longblob,text,mediumtext,longtext"
+- 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。使用 `ANALYZE TABLE t COLUMNS c1, ..., cn` 语法时，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，则不会收集该列的统计信息。
```
Comment (songrijie) [inline system-variables.md:848]:
```suggestion
- 这个变量表示在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。即使使用 `ANALYZE TABLE t COLUMNS c1, ..., cn` 语法指定列，如果指定的列的类型在 `tidb_analyze_skip_column_types` 中，也不会收集该列的统计信息。
```

File: statistics.md [hunk 11]
```diff
@@ -131,7 +131,8 @@ ANALYZE TABLE TableNameList [WITH NUM BUCKETS|TOPN|CMSKETCH DEPTH|CMSKETCH WIDTH
 
 > **注意：**
 >
-> 收集部分列的统计信息的功能仅适用于 `tidb_analyze_version = 2` 的情况。
+> 收集部分列的统计信息的功能仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。
+> TiDB v7.2.0 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，该变量可以控制在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```
Comment (ran-huang) [inline statistics.md:135]:
```suggestion
> - 收集部分列的统计信息的功能仅适用于 [`tidb_analyze_version = 2`](#tidb_analyze_version-从-v510-版本开始引入) 的情况。
> - TiDB v7.2.0 引入了系统变量 [`tidb_analyze_skip_column_types`](/system-variables.md#tidb_analyze_skip_column_types-从-v720-版本开始引入)，该变量可以控制在执行 `ANALYZE` 命令收集统计信息时，跳过哪些类型的列的统计信息收集。该变量仅适用于 `tidb_analyze_version = 2` 的情况。
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**